### PR TITLE
refactor(add):  strip `.ts` and `.vue` extension from name

### LINKED ISSUE #175: make nuxi add detect and accomodate extensions
❯ npx nuxi add component blog/BlogFeeds.vue
ℹ 🪄 Generated a new component in /Users/k/vc/web/components/blog/BlogFeeds.vue.vue                                      4:46:50 PM
❯ npx nuxi add component blog/BlogFeeds
ℹ 🪄 Generated a new component in /Users/k/vc/web/components/blog/BlogFeeds.vue       

Detect that I supplied the `.vue` and don't automatically add it?

### PR DESCRIPTION
**Pull request by shinGangan**
### 🔗 Linked issue

closed #175  .

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Fixed nuxi add command to check arg.name extension.

```sh
❯ npx nuxi add component blog/BlogFeeds.vue
ℹ 🪄 Generated a new component in /Users/k/vc/web/components/blog/BlogFeeds.vue
❯ npx nuxi add component blog/BlogFeeds
ℹ 🪄 Generated a new component in /Users/k/vc/web/components/blog/BlogFeeds.vue
❯ npx nuxi add component Modal/index
ℹ 🪄 Generated a new component in /Users/k/vc/web/components/Modal/index.vue
```

- Non Implementation 1: use of `endsWith`
- Non implementation 2: Checking for anything other than `.ts`, `.vue`
  - Users who want to use `nuxi add` command often want to generate `.vue` or `.ts` files.
  - If the number of extensions to be checked increases, the following is also acceptable

    ```js
    const exts = ['.ts', '.vue']
    function checkExt(name: string) {
      for (let i=0; i < exts.length; i++) {
        if(name.endsWith(exts[i])) {
          return name.replace(f, '')
        }
      }
      return name
    }

    /***/
    const name = checkExt(ctx.args.name)
    ```

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have added tests (if possible).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -80,6 +80,6 @@
   },
   "packageManager": "pnpm@8.15.2",
   "engines": {
-    "node": "^14.18.0 || >=16.10.0"
+    "node": "^16.10.0 || >=18.0.0"
```
Comment (shinGangan) [inline package.json:83]: ### Note

This update is based on the following issue and comments.

- https://github.com/nuxt/nuxt/pull/23821
- https://github.com/nuxt/nuxt/pull/23821#pullrequestreview-1689862898

**Comment by shinGangan** [comment]
The review has a low priority. Please review when you have a time.
Thank you for your reviews everytime. 🙏

File: package.json (hunk 1 above)
Comment (pi0) [reply package.json:83]: Looks good but would you mind to split this PR? 🙏🏼

File: src/commands/add.ts [hunk 2]
```diff
@@ -33,7 +33,12 @@ export default defineCommand({
     const cwd = resolve(ctx.args.cwd || '.')
 
     const template = ctx.args.template
-    const name = ctx.args.name
+    const name =
+      extname(ctx.args.name) === '.vue'
```
Comment (pi0) [inline src/commands/add.ts:37]: Refactor, in line above, you can use `const ext = extname(ctx.args.name)`

File: package.json (hunk 1 above)
Comment (shinGangan) [reply package.json]: OK 🙆
This content has been split into a PR here.

- nuxt/cli#356

**Review by pi0 — APPROVED** [review]

**Comment by shinGangan** [comment]
@pi0 , thanks for support for me 🙏

**Pull request merged**